tests/scheduled_task/scheduled_task_suite_test.go

```diff
@@ -0,0 +1,237 @@
+// Copyright DataStax, Inc.
+// Please see the included license file for details.
+
+package scheduled_task
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/k8ssandra/cass-operator/tests/kustomize"
+	ginkgo_util "github.com/k8ssandra/cass-operator/tests/util/ginkgo"
+	"github.com/k8ssandra/cass-operator/tests/util/kubectl"
+)
+
+var (
+	testName              = "ScheduledTask"
+	namespace             = "test-scheduled-task"
+	dcName                = "dc1"
+	dcYaml                = "../testdata/default-scheduled-task-dc.yaml"
+	scheduledTaskYaml     = "../testdata/tasks/scheduled_cleanup_task.yaml"
+	scheduledTaskName     = "scheduled-cleanup-task"
+	dcResource            = fmt.Sprintf("CassandraDatacenter/%s", dcName)
+	scheduledTaskResource = fmt.Sprintf("ScheduledTask/%s", scheduledTaskName)
+	dcLabel               = fmt.Sprintf("cassandra.datastax.com/datacenter=%s", dcName)
+	ns                    = ginkgo_util.NewWrapper(testName, namespace)
+)
+
+func TestLifecycle(t *testing.T) {
+	AfterSuite(func() {
+		logPath := fmt.Sprintf("%s/aftersuite", ns.LogDir)
+		err := kubectl.DumpAllLogs(logPath).ExecV()
+		if err != nil {
+			t.Logf("Failed to dump all the logs: %v", err)
+		}
+
+		fmt.Printf("\n\tPost-run logs dumped at: %s\n\n", logPath)
+		ns.Terminate()
+		err = kustomize.Undeploy(namespace)
+		if err != nil {
+			t.Logf("Failed to undeploy cass-operator: %v", err)
+		}
+	})
+
+	RegisterFailHandler(Fail)
+	RunSpecs(t, testName)
+}
+
+var _ = Describe(testName, func() {
+	Context("when in a new cluster", func() {
+		Specify("the operator can schedule and execute CassandraTasks", func() {
+			var step string
+			var json string
+			var k kubectl.KCmd
+
+			By("deploy cass-operator with kustomize")
+			err := kustomize.Deploy(namespace)
+			Expect(err).ToNot(HaveOccurred())
+
+			step = "waiting for operator pod to be created"
+			json = "jsonpath={.items[*].metadata.name}"
+			k = kubectl.Get("pods").
+				WithLabel("name=cass-operator").
+				FormatOutput(json)
+			ns.WaitForOutputContainsAndLog(step, k, "cass-operator", 300)
+
+			ns.WaitForOperatorReady()
+
+			step = "creating a datacenter resource with 2 nodes"
+			testFile, err := ginkgo_util.CreateTestFile(dcYaml)
+			Expect(err).ToNot(HaveOccurred())
+
+			k = kubectl.ApplyFiles(testFile)
+			ns.ExecAndLog(step, k)
+
+			ns.WaitForDatacenterReady(dcName)
+
+			step = "creating a ScheduledTask for cleanup"
+
+			k = kubectl.ApplyFiles(scheduledTaskYaml)
+			stdout, stderr, err := ns.ExecVCapture(k)
+			Expect(err).ToNot(HaveOccurred())
+			fmt.Println(stdout)
```

Comment:
No need for fmt.Println, logging will take care of these when used. ns.ExecAndLog(step, k)